### PR TITLE
Astropy v5

### LINKED ISSUE #59: Move to Astropy v5
Keeping up with the Jones' etc.

Also add type annotations for Quantities.

### PR DESCRIPTION
**Pull request by steven-murray**
Fixes #59 

This PR does the following:

* Changes all inputs so that they *require* astropy quantities (where applicable)
* Adds a `yaml` module that helps to read the configuration files with variables specified as quantities.
* Adds type annotations everywhere, including the new astropy unit-aware types.

This is intended to make everything consistent for the user experience, but also to make it possible to infer parameter properties directly by inspecting the parameters themselves. This will be helpful for UI's built on top of 21cmSense.

### CODE REVIEW

File: .github/workflows/testsuite.yaml [hunk 1]
```diff
@@ -40,6 +40,7 @@ jobs:
       - name: Run Tests
         run: |
           pip install .[test]
+          conda list
```
Comment (mkolopanis) [inline .github/workflows/testsuite.yaml:43]: The actual test output can be quite verbose. I wonder if the install and conda list should be a separate step?

**Review by mkolopanis — COMMENTED** [review]
This is looking pretty nice. I do have two small question/comments.

File: .github/workflows/testsuite.yaml (hunk 1 above)
Comment (steven-murray) [reply .github/workflows/testsuite.yaml:43]: Good point. I've moved it to its own step.

File: py21cmsense/conversions.py [hunk 2]
```diff
@@ -6,15 +6,18 @@
 import numpy as np
 from astropy import constants as cnst
 from astropy import units as u
```
[the remark this answers is not in the document]
Comment (steven-murray) [reply py21cmsense/conversions.py]: I've made it `un` everywhere.

**Review by mkolopanis — APPROVED** [review]
Thanks for the little changes!

**Pull request merged**